akka-http/src/main/scala/akka/http/scaladsl/server/directives/FileUploadDirectives.scala

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.http.scaladsl.server.directives
+
+import java.io.File
+
+import akka.http.scaladsl.server.{ MissingFormFieldRejection, Directive1 }
+import akka.http.scaladsl.model.{ ContentType, Multipart }
+import akka.util.ByteString
+import scala.concurrent.Future
+import scala.util.{ Failure, Success, Try }
+import akka.stream.scaladsl._
+
+trait FileUploadDirectives {
+
+  import BasicDirectives._
+  import RouteDirectives._
+  import FutureDirectives._
+  import MarshallingDirectives._
+
+  /**
+   * Streams the bytes of the file submitted using multipart with the given file name into a temporary file on disk.
+   * If there is an error writing to disk the request will be rejected, if there is no such field the request will
```

Comment:
Last nitpick: this is not implemented as rejecting, it uses `failWith` which is "will be failed with the cause exception".
Reject is "this route does not match".